apps/admin/frontend/src/components/print_diagnostic_button.tsx

```diff
@@ -0,0 +1,134 @@
+import { assert, assertFalsy, sleep } from '@votingworks/basics';
+import type { DiagnosticsOutcome } from '@votingworks/admin-backend';
+import { addDiagnosticRecord, getPrinterStatus, printTestPage } from '../api';
+import { Loading } from './loading';
+
+/**
+ * Normally we want to free up the UI sooner, before the print is actually
+ * done, but in this case we want to wait until the print is done or almost
+ * done before presenting the verification step.
+ */
+export const TEST_PAGE_PRINT_DELAY_SECONDS = 7;
+
+type FlowState = 'printing' | 'verification' | 'test-failed';
+
+export function PrintTestPageButton(): JSX.Element {
+  const [flowState, setFlowState] = useState<FlowState>();
+  const [outcome, setOutcome] = useState<DiagnosticsOutcome>();
+
+  const printerStatusQuery = getPrinterStatus.useQuery();
+  const printTestPageMutation = printTestPage.useMutation();
+  const addDiagnosticRecordMutation = addDiagnosticRecord.useMutation();
+
+  const isPrinterConnected =
+    printerStatusQuery.isSuccess && printerStatusQuery.data.connected;
+
+  function resetFlow() {
+    setFlowState(undefined);
+    setOutcome(undefined);
+    printTestPageMutation.reset();
+    addDiagnosticRecordMutation.reset();
+  }
+
+  async function startFlow() {
+    printTestPageMutation.mutate();
+    setFlowState('printing');
+    await sleep(TEST_PAGE_PRINT_DELAY_SECONDS * 1000);
+    setFlowState('verification');
+  }
+
+  function verifyPrintOutcome() {
+    assert(outcome !== undefined);
+
+    addDiagnosticRecordMutation.mutate({
+      hardware: 'printer',
+      outcome,
+    });
+
+    if (outcome === 'fail') {
+      setFlowState('test-failed');
+    } else {
+      resetFlow();
+    }
+  }
+
+  let modal: JSX.Element | null = null;
+
+  switch (flowState) {
+    case 'printing':
+      modal = (
+        <Modal centerContent content={<Loading>Printing Test Page</Loading>} />
+      );
+      break;
+    case 'verification':
+      modal = (
+        <Modal
+          title="Test Page Printed"
+          actions={
+            <React.Fragment>
+              <Button
+                onPress={verifyPrintOutcome}
+                disabled={!outcome}
+                variant="primary"
+              >
+                Confirm
+              </Button>
+              <Button onPress={resetFlow}>Cancel</Button>
+            </React.Fragment>
+          }
+          content={
+            <React.Fragment>
+              <P>Inspect the test page and select a test outcome:</P>
+              <RadioGroup
+                label="Test Result"
+                hideLabel
+                value={outcome}
+                options={[
+                  {
+                    value: 'pass',
+                    label: 'Pass - printed without print quality issues',
+                  },
+                  {
+                    value: 'fail',
+                    label:
+                      'Fail - no print or printed with a print quality issue',
```

Comment:
Good point. I only reproduce it when I'm in "device emulation mode" in dev tools (the video is screen clipped) so maybe it's something that happens in response to touch? I would be surprised if that were a problem generally because @kofi-q would have seen it in VxMark. So I'm guessing it's something about my dev tools setup, or the radio component's placement in a modal?

Reply:
Weird, haven't seen that before either -- almost looks like one of those overlays chrome dev tools puts over elements when highlighted in the elements panel, but wouldn't make sense for it to be triggered here

Reply:
Hmm, yeah it seemed like a dev tools thing to me as well